models/base_model.py

```diff
@@ -0,0 +1,160 @@
+import os
+import torch
+from collections import OrderedDict
+from . import networks
+
+
+class BaseModel:
+
+    # modify parser to add command line options,
+    # and also change the default values if needed
+    @staticmethod
+    def modify_commandline_options(parser, is_train):
+        return parser
+
+    def name(self):
+        return "BaseModel"
+
+    def initialize(self, opt):
+        self.opt = opt
+        self.use_gpu = opt.model.use_gpu
+        self.device = torch.device(
+            "cuda" if (torch.cuda.is_available() and opt.model.use_gpu) else "cpu"
+        )
+        self.isTrain = opt.model.is_train
+        self.loss_names = []
+        self.model_names = []
+        self.comet_exp = opt.comet_exp
+        self.save_dir = opt.train.output_dir + "/checkpoints"
+
+    def set_input(self, input):
+        pass
```

Comment:
It's a detail but if the meaning of these `pass` statements is to say that this must be implemented by classes which inherit from `BaseModel` then you should have `raise NotImplementedError("implement set_input")` instead